README.md

```diff
@@ -316,99 +625,104 @@ Before using the `candidate_ranker` module of DeezyMatch, we need to:
+Here, `rnn_passes` specifies that `combine_vecs` should assemble all vectors generated in the forward and backward RNN/GRU/LSTM passes (and stored in the `input_scenario` directory). NOTE: we have a backward pass only if `bidirectional` is set to `True` in the input file.
 
-Similarly, this can be done via command line:
+The results (for both query and candidate vectors) are sored in the `output_scenario` as follows:
 
 ```bash
-DeezyMatch --deezy_mode combine_vecs -qc q,c -p fwd,bwd -sc test -combs test
+combined/
+├── candidates_test
+│   ├── bwd_id.pt
+│   ├── bwd_items.npy
+│   ├── bwd.pt
+│   ├── fwd_id.pt
+│   ├── fwd_items.npy
+│   ├── fwd.pt
+│   └── input_dfm.yaml
+└── queries_test
+    ├── bwd_id.pt
+    ├── bwd_items.npy
+    ├── bwd.pt
+    ├── fwd_id.pt
+    ├── fwd_items.npy
+    ├── fwd.pt
+    └── input_dfm.yaml
 ```
 
-In this command, compared to `combine_vecs` module explained above:
-* `-qc`: `qc_modes`
-* `-p`: `rnn_passes`
-* `-sc`: `input_scenario`
-* `-combs`: `output_scenario`
+The above steps can be done via command line, for query vectors:
+
+```bash
+DeezyMatch --deezy_mode combine_vecs -p fwd,bwd -sc queries/test -combs combined/queries_test
+```
 
-The results are sored in the `output_scenario` (in python module mode) or `-combs` (in command-line mode) as follows:
+For candidate vectors:
 
 ```bash
-combined
-└── test
-    ├── candidates_bwd.pt
-    ├── candidates_bwd_id.pt
-    ├── candidates_bwd_items.npy
-    ├── candidates_fwd.pt
-    ├── candidates_fwd_id.pt
-    ├── candidates_fwd_items.npy
-    ├── input_dfm.yaml
-    ├── queries_bwd.pt
-    ├── queries_bwd_id.pt
-    ├── queries_bwd_items.npy
-    ├── queries_fwd.pt
-    ├── queries_fwd_id.pt
-    ├── queries_fwd_items.npy
-    └── test_candidates_deezymatch.pkl
+DeezyMatch --deezy_mode combine_vecs -p fwd,bwd -sc candidates/test -combs combined/candidates_test
 ```
 
+In this command, compared to `combine_vecs` module explained above:
+* `-p`: `rnn_passes`
+* `-sc`: `input_scenario`
+* `-combs`: `output_scenario`
+
 #### CandidateRanker
 
-Various options are available to find a set of candidates (from a dataset) for a given query in the same or another dataset.
+Candidate ranker uses the vector representations, generated and assembled in the previous sections, to find a set of candidates (from a dataset) for given queries in the same or another dataset. In the following example, for queries stored in `query_scenario`, we want to find 2 candidates (specified by `num_candidates`) from a dataset stored in `candidate_scenario`.
 
-* Select candidates based on L2-norm distance (aka faiss distance):
+:warning: It is also possible to do [candidate ranking on-the-fly](#candidate-ranking-on-the-fly) in which query vectors are generated on-the-fly (and not stored in a dataset).
```

Comment:
The phrase "(and not stored in a dataset)" can be ambiguous, because the output of candidate ranking on-the-fly is stored in a dataset. I guess you mean that the query vectors are not read from a dataset (instead, they are generated on the fly)?

Reply:
Thanks, done.